src/main/java/uk/gov/pay/products/service/PaymentFinder.java

```diff
@@ -27,34 +27,46 @@ public PaymentFinder(PaymentDao paymentDao, LinksDecorator linksDecorator, Publi
         this.publicApiRestClient = publicApiRestClient;
     }
 
-    @Transactional
     public List<Payment> findByProductExternalId(String productExternalId) {
-        List<Payment> payments = paymentDao.findByProductExternalId(productExternalId)
+        return findPaymentEntitiesByProductExternalId(productExternalId)
                 .stream()
-                .map(paymentEntity -> linksDecorator.decorate(paymentEntity.toPayment()))
+                .map(paymentEntity -> {
+                    Payment payment = queryGovUKPaymentStatus(paymentEntity);
+                    return linksDecorator.decorate(payment);
```

Comment:
What happens if one of the payment throws an error here?

Reply:
Added try/catch it logs and swallows error